pixeltable/tests/test_dataframe.py

```diff
@@ -25,6 +31,34 @@ def test_select_where(self, test_tbl: catalog.Table) -> None:
             _ = t.select(datetime.datetime.now).show(0)
         assert 'Not a valid literal' in str(exc_info.value)
 
+        # catch invalid alias in select list from user input
+        # only check stuff that's not caught by python kwargs checker
+        with pytest.raises(exc.Error) as exc_info:
+            _ = t.select(t.c1, **{'c2-1': t.c2}).show(0)
+        assert 'Invalid alias' in str(exc_info.value)
+
+        with pytest.raises(exc.Error) as exc_info:
+            _ = t.select(t.c1, **{'': t.c2}).show(0)
+        assert 'Invalid alias' in str(exc_info.value)
+
+        with pytest.raises(exc.Error) as exc_info:
+            _ = t.select(t.c1, **{'foo.bar': t.c2}).show(0)
+        assert 'Invalid alias' in str(exc_info.value)
+
+        # catch repeated alias from user input
+        with pytest.raises(exc.Error) as exc_info:
+            _ = t.select(t.c2, c2=t.c1).show(0)
+        assert 'Repeated alias' in str(exc_info.value)
+
+        with pytest.raises(exc.Error) as exc_info:
+            _ = t.select(t.c2, t.c2).show(0)
```

Comment:
Hm, this should probably be legal.

In the end, the alias is simply meant for displaying the output, so maybe duplicates don't matter.

Reply:
Current version makes it legal, but makes a new name for the repeated column to keep names unique. 
There's a test checking this is legal.